src/makielayout/blocks/intervalslider.jl

```diff
@@ -178,47 +178,42 @@ function initialize_block!(isl::IntervalSlider)
                 isl.selected_indices[] = newindices
             end
         end
+    end
 
+    onmouseleftdrag(mouseevents) do event
+        handle_event(event)
         return Consume(true)
     end
 
     onmouseleftdragstop(mouseevents) do event
-        dragging[] = false
+        isl.dragging[] = false
         # adjust slider to closest legal value
         sliderfractions[] = sliderfractions[]
+        notify(isl.selected_indices)
         return Consume(true)
     end
 
     onmouseleftdown(mouseevents) do event
+        isl.dragging[] = true
 
-        pos = event.px
-
-        dim = isl.horizontal[] ? 1 : 2
-        frac = clamp(
-            (pos[dim] - endpoints[][1][dim]) / (endpoints[][2][dim] - endpoints[][1][dim]),
-            0, 1
-        )
-
-        startfraction[] = frac
+        fraction = get_fraction(event)
+        startfraction[] = fraction
         startindices[] = isl.selected_indices[]
         start_disp_fractions[] = displayed_sliderfractions[]
 
-        if state[] in (:both, :none)
-            return Consume(true)
-        end
+        handle_event(event)
 
-        newindex = closest_fractionindex(isl.range[], frac)
-        if abs(newindex - isl.selected_indices[][1]) < abs(newindex - isl.selected_indices[][2])
-            isl.selected_indices[] = (newindex, isl.selected_indices[][2])
-        else
-            isl.selected_indices[] = (isl.selected_indices[][1], newindex)
-        end
-        # linecolors[] = [color_active[], color_inactive[]]
+        return Consume(true)
+    end
 
+    onmouseleftclick(mouseevents) do event
+        isl.dragging[] = false
+        notify(isl.selected_indices)
         return Consume(true)
     end
```

Comment:
Clicks also trigger a drag start so you end up with a duplicate update here too:
```
IntervalSlider drag true
IntervalSlider interval (0.43, 10.0)
IntervalSlider indices (44, 1001)
[ Info: click start
IntervalSlider drag false
IntervalSlider interval (0.43, 10.0)
IntervalSlider indices (44, 1001)
[ Info: click end
```

There is an exception to this her though: If you do not move the slider through a click you do not get an interval update through the drag start. So then you only get the click update, which however does not change the values. I'm not sure if that's an update worth pushing out.